File: SDK-v5/custom-gcmlistenerservice/app/src/main/java/com/localytics/android/sample/gcmlistenerservice/MyInstanceIDListenerService.java
package com.localytics.android.sample.gcmlistenerservice;

import android.content.Intent;

import com.google.android.gms.iid.InstanceIDListenerService;

public class MyInstanceIDListenerService extends InstanceIDListenerService
{
    @Override
    public void onTokenRefresh() {
        Intent intent = new Intent(this, RegistrationIntentService.class);
        startService(intent);
    }
}

File: SDK-v5/kotlin/custom-gcmlistenerservice/app/src/main/java/com/localytics/android/sample/gcmlistenerservice/MyInstanceIDListenerService.kt
package com.localytics.android.sample.gcmlistenerservice

import android.content.Intent

import com.google.android.gms.iid.InstanceIDListenerService

class MyInstanceIDListenerService : InstanceIDListenerService() {
    override fun onTokenRefresh() {
        val intent = Intent(this, RegistrationIntentService::class.java)
        startService(intent)
    }
}
